python/kserve/kserve/protocol/rest/openai.py

```diff
@@ -0,0 +1,172 @@
+import time
+from typing import Dict, List, Literal, Optional, Union
+
+from pydantic import BaseModel, Field
+import torch
+
+def random_uuid() -> str:
+    return str(uuid.uuid4().hex)
+
+class ErrorResponse(BaseModel):
+    object: str = "error"
+    message: str
+    type: str
+    param: Optional[str] = None
+    code: int
+
+class UsageInfo(BaseModel):
+    prompt_tokens: int = 0
+    total_tokens: int = 0
+    completion_tokens: Optional[int] = 0
+
+class ChatCompletionRequest(BaseModel):
```

Comment:
Let's use the `openai.types` like in the other PR?
https://github.com/kserve/kserve/pull/3477/files#diff-308b33d915f065d9988e8c7f2d31862a433ec80c941149573d2114e9fea2bce9R12

Reply:
I see there will be changes of duplicate changes. Do we know when will `OpenAIModel()` plan to be complete?